Draft of DIDComm protocol

TelegramSam opened the pull request.

TelegramSam committed:
Draft of DIDComm protocol

ArPhil commented:
Thanks @TelegramSam !!

As far as I understood, the binding (you call it association) of the token with the DIDComm connection (and therefore the token holder DID and issuer (receiver?) DID) is done, by presenting the token and its confirmation.

Since this is a very generic flow which describes the association of any (bearer?) token with a DIDComm association / connection, I am thinking whether the proposed DIDComm protocol should emhpasize its openess even more? You mention, that it is primarly meant for OpenID4VC interactions but that it can be used for any OID interactions. But actually I believe this holds true for other (access) token based protocols as well?

~~When exactly do you want to use it within our "OpenIDIDComm" Flow?~~

Happy to discuss this tomorrow!

TelegramSam commented:
Any reason to not merge this?

ArPhil commented on Code/diddcomm_protocol.md:
```diff
@@ -0,0 +1,60 @@
+
+This is a protocol that associates a DIDComm relationship with an access token obtained via OID interactions. While the anticipated use of this protocol is alongsid OID4VC protocol interactions, it can be used with any OID interactions.
+
+This protocol plays a small but important part in the overall flow that associates OID and DIDComm interactions. For full details, see <link here>. Prior to use of this protocol, a DIDComm associated access token was obtained from OID or OID4VC interactions. 
```
Which details do you want to link in `<link here>`? Not sure whether I understand the following statement entirely.

ArPhil replied:
@TelegramSam ,

I pushed another commit with some minor changes regarding the roles (added *token_verifier*). The reason why I believe, that we should make this differentiation lies (as @void1042 probably would say 😉 ) "in the spirit of OAuth":

There we have the separation of the authorization server (which issues the token to the client) and the resource server (which is not necessarily the same entity as the authorization server) and grants access to the protected ressource by verifiyng it. In this case I see the *DIDComm channel* as the protected ressource.

As long as what I wrote makes sense to you, I am more than happy to merge the PR. :)

ArPhil committed:
Update diddcomm_protocol.md

Differentiated token_issuer and token_holder roles, though it most likely will be the same entity.
